### PR TITLE
Use `docker compose` instead of `docker-compose`

### LINKED ISSUE #1449: Only use `docker compose` instead of `docker-compose`
### Is your feature request related to a specific problem?

`docker-compose` is no longer the preferred means of running docker compose functionality - `docker compose` (lack of hyphen) is. what's more - the minimum version of docker that we're now requiring necessarily includes support for `docker compose`, so we should be safe to rely on it. 

we still have some references to `docker-compose` hanging around in our nox files, and we also run a check on the `docker-compose` version. while this _may_ resolve to the same thing as `docker compose` for developers, depending on a value in their docker settings, there's a chance it doesn't, in which case it will add unneeded confusion and potential issues.

### Describe the solution you'd like

- update all references to `docker-compose` to instead use `docker compose`
- everything _should_ just work, i think? but obviously, make sure that's the case...

### Additional context

this isn't urgent, so if for some reason it's not a no-brainer, then no need to rush it.

### PR DESCRIPTION
**Pull request by adamsachs**
Closes #1449 

### Code Changes

* Use `docker compose` instead of `docker-compose`
* We also no longer need to explicitly check the `docker-compose` version - checking the general `docker` version should suffice.

### Steps to Confirm

* Make sure all our nox sessions still work as expected locally
* Make sure all our CI checks are not negatively impacted

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

See description of #1449 for more detail

### CODE REVIEW

**Comment by adamsachs** [comment]
@ThomasLaPiana pretty sure the test failure here is unrelated - but not sure if you've got ideas on whether we should do anything else here to verify this won't break anything?

File: docs/fides/docs/tutorial/add.md
Comment (ThomasLaPiana) [inline docs/fides/docs/tutorial/add.md:47]:
```suggestion
Now that the dependency is included in the project and the configuration is in place, the fides server needs to be told to run. The app uses `docker compose` to orchestrate resources, so include `fides` as a service by adding the following configuration after the database service:
```

**Comment by ThomasLaPiana** [comment]
@adamsachs I merged in main and move the compose check to the top of the nox startup, so we don't have to remember to run it in certain commands. This broke the static checks it looks like so I'll get those sorted and then hand it back to you for a once-over

**Comment by ThomasLaPiana** [comment]
got most things running, but for some reason fides checks don't want to run, but I'm pretty sure its the same container 🤔

**Review by ThomasLaPiana — APPROVED** [review]

**Comment by ThomasLaPiana** [comment]
@adamsachs thank you for popping this open!

**Comment by ThomasLaPiana** [comment]
@sanders41 can you give this a once-over and merge if everything looks ok? Just want another set of eyes before I ship it

**Pull request merged**
